libr/bin/p/bin_hunk.c

```diff
@@ -84,17 +61,31 @@ static RList* sections(RBinFile *bf) {
 	return ret;
 }
 
-static RList* entries(RBinFile *bf) { //Should be 3 offsets pointed by NMI, RESET, IRQ after mapping && default = 1st CHR
+static RList* entries(RBinFile *bf) {
 	RList *ret;
 	RBinAddr *ptr = NULL;
+	RBuffer *buf;
+	int addr;
+	ut8 b[4];
 	if (!(ret = r_list_new ())) {
 		return NULL;
 	}
 	if (!(ptr = R_NEW0 (RBinAddr))) {
 		return ret;
 	}
-	ptr->paddr = 0x24;
-	ptr->vaddr = 0x24;
+	buf = bf->buf;
+	for (addr = 0x18; addr <= (r_buf_size (buf) - 4); addr += 0x4) {
+		r_buf_read_at (buf, addr, b, sizeof (b));
+		if (!memcmp (b, HUNK_CODE, sizeof (b))) {
```

Comment:
Uhm thats pretty slow. What about read once a 64kb block (orbrestrict to a sane limit and use r_mem_mem and check if the result is aligned to 4? Or just do the read once avoiding the interation on large files and keep the memcmp loop

Reply:
the hunk parser is still very naive, so for this kind of parser 1k read will be enough. im planning proper hunk parsing for iS (section info) anyway, but i wanna keep this PR here small and fix more urgent issue here first.